nixos/modules/misc/wordlist.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+with lib;
+{
+  options = {
+    environment.wordlist = {
+      enable = mkEnableOption "environment variables for lists of words";
+
+      lists = mkOption {
+        type = types.attrsOf (types.nonEmptyListOf types.path);
+
+        default = {
+          WORDLIST = [ "${pkgs.scowl}/share/dict/words.txt" ];
+        };
+
+        defaultText = literalExpression ''
+          {
+            WORDLIST = [ "''${pkgs.scowl}/share/dict/words.txt" ];
+          }
+        '';
+
+        description = ''
+          A set with the key names being the environment variable you'd like to
+          set and the values being a list of paths to text documents containing
+          lists of words. A concatenation of the files will be created and
+          stored in the nix store.
+
+          If you have a handful of words that you want to add to an already
+          existing wordlist, you may find `pkgs.writeText` useful for this task.
+        '';
+
+        example = literalExpression ''
+          {
+            WORDLIST = [ "''${pkgs.scowl}/share/dict/words.txt" ];
+            AUGMENTED_WORDLIST = [
+              "''${pkgs.scowl}/share/dict/words.txt"
+              "''${pkgs.scowl}/share/dict/words.variants.txt"
+              "''${pkgs.writeText "extra-words" (lib.concatStringsSep "\n" [ "desynchronization" "oobleck" ""])}"
+              # Final empty string to get a trailing newline, important for composing multiple lists.
```

Comment:
```suggestion
              (builtins.toFile "extra-words" ''
                desynchronization
                oobleck
              '')
```

Reply:
> we just tested simple concatenation of wordlists for `look` (which seems the main use case in the linked issue?), and it doesn't work. confused `look` requires the word list to be sorted to function, so this module has to sort its word lists as well. (and if we have to use more complicated methods of building the final wordlist we can also insert a `\n` between two files, obviating the need for a trailing `\n` on every file)
> 
> apart from that this module does look good now though.

Sorting means that I can't use the built-in `pkgs.concatText` but that's OK.  I'll just go back to `pkgs.runCommand` instead, and add a few better ways to make sure that the whitespace is correct (and there aren't doubled entries and so on).

Reply:
> we just tested simple concatenation of wordlists for `look` (which seems the main use case in the linked issue?), and it doesn't work. confused `look` requires the word list to be sorted to function, so this module has to sort its word lists as well. (and if we have to use more complicated methods of building the final wordlist we can also insert a `\n` between two files, obviating the need for a trailing `\n` on every file)
> 
> apart from that this module does look good now though.

No worries, I just think this means I can't use `pkgs.concatText` anymore, but that's not a big deal.  It also gives me the chance to make damn sure that all of the whitespace is right, and there's no leading/trailing space and there's no empty lines.  `awk 1` is there to concatenate the files while adding newlines between files if needed.  Fewer footguns.